index.js

```diff
@@ -1986,14 +1988,31 @@ export function generate(options) {
     let rightSize = false;
     let wordUsed;
     while (!rightSize) {
-      wordUsed = generateRandomWord();
+      // Generate a random word with the provided prefix (if any)
+      wordUsed = generateRandomWord(prefix);
```

Comment:
While it was already possible in this loop, adding `prefix` increases the likelihood that we'll never find a word and get stuck in an infinite loop, which is problematic for sure. So please refactor `generateRandomWord` to also take `min` and `max` and include them in your `filter` criteria. That way we will know if it's impossible to satisfy the request due to any combination of `min`, `max` and `prefix` and can throw one clean exception. And this "rightSize" while loop can go away.

Probably generateRandomWord should just be merged with this function.